src/cloudWatchLogs/activation.ts

```diff
@@ -39,11 +39,21 @@ export async function activate(context: vscode.ExtensionContext, configuration:
     context.subscriptions.push(
         vscode.workspace.onDidCloseTextDocument(doc => {
             if (doc.isClosed && doc.uri.scheme === CLOUDWATCH_LOGS_SCHEME) {
+                registry.disposeRegistryData(doc.uri)
                 registry.deregisterLog(doc.uri)
```

Comment:
off-topic (future PR?): should deregisterLog be renamed? Does "log" mean "logstream" or "logdata" or "loggroup" ...?

Do we even need `deregisterLog()` anymore? It is only called where `disposeRegistryData()` is called. So its job should be merged into `disposeRegistryData()`.

Reply:
I think you're right. These two function have a very similar function and can realistically always be called together. I went ahead and combined them into `disposeRegistryData()`. 

This seemed like an opportunity to reduce code volume which I assume is always a good goal to have in mind?